build/pause/windows/pause.c

```diff
@@ -0,0 +1,35 @@
+#include <windows.h>
+#include <stdio.h>
+
+
+BOOL WINAPI CtrlHandler(DWORD fdwCtrlType)
+{
+	switch (fdwCtrlType)
+	{
+	case CTRL_C_EVENT:
+		fprintf(stderr, "Shutting down, got signal\n");
+		exit(0);
+
+	case CTRL_BREAK_EVENT:
+		fprintf(stderr, "Shutting down, got signal\n");
+		exit(0);
+
+	default:
+		return FALSE;
+	}
+}
+
+int main(void)
+{
+	if (SetConsoleCtrlHandler(CtrlHandler, TRUE))
+	{
+		while (1) {}
```

Comment:
this infinite loop can consume a full hardware thread (e.g. if running this application on the host).
we want `Sleep(INFINITE);` instead of:
```
while (1) {}
```